src/browser/api/window.js

```diff
@@ -1029,15 +1029,21 @@ Window.getGroup = function(identity) {
 
 
 Window.getWindowInfo = function(identity) {
-    let browserWindow = getElectronBrowserWindow(identity, 'get info for');
-    let webContents = browserWindow.webContents;
-
-    return {
-        url: webContents.getURL(),
-        title: webContents.getTitle(),
+    const browserWindow = getElectronBrowserWindow(identity, 'get info for');
+    const openfinWindow = Window.wrap(identity.uuid, identity.name);
+    const webContents = browserWindow.webContents;
+    const windowInfo = {
+        canNavigateBack: webContents.canGoBack(),
         canNavigateForward: webContents.canGoForward(),
-        canNavigateBack: webContents.canGoBack()
+        title: webContents.getTitle(),
+        url: webContents.getURL()
     };
+
+    if (openfinWindow.preloadState) {
+        windowInfo.preloadState = openfinWindow.preloadState;
```

Comment:
Should preload state always be included in the payload? Is there a default value that could represent this? `null`?